.github/workflows/generation.yml

```diff
@@ -34,53 +41,46 @@ jobs:
             --additional-properties=modelPackage=models
             --additional-properties=apiPackage=api
             --enable-post-process-file
-            -o /github/workspace/unstable
-
-      - name: Get server commit hash
-        id: sha
-        run: |
-          SERVER_SHA="$(git ls-remote git://github.com/jellyfin/jellyfin.git | grep refs/heads/master | cut -f 1)"
-          echo "::set-output name=sha::$(echo ${SERVER_SHA::7})"
+            -o /github/workspace/${{ matrix.version }}
 
       - name: Check if .ts files were modified
         id: diff
-        run: echo "::set-output name=count::$(git status -s | grep ts | wc -l)"
+        run: echo "::set-output name=count::$(git status -su | grep ts | wc -l)"
 
-      - name: Get OpenApi version
-        if: ${{ steps.diff.outputs.count > 0 }}
-        id: apiversion
+      - name: (stable) Set version in package.json
+        if: ${{ steps.diff.outputs.count > 0 && matrix.version == 'stable' }}
         run: |
-          SERVER_VERSION="$(cat openapi.json | jq '.info.version')"
-          echo "::set-output name=number::$(echo ${SERVER_VERSION::-1})"
+          VERSION=$(cat openapi.json | jq '.info.version')
+          echo "$(cat stable/package.json | jq ".version = $VERSION")" > stable/package.json
+          echo "$(cat stable/package-lock.json | jq ".version = $VERSION")" > stable/package-lock.json
 
-      - name: Create new version string
-        if: ${{ steps.diff.outputs.count > 0 }}
-        id: version
-        run: echo "::set-output name=number::$(echo ${{ steps.apiversion.outputs.number }}-unstable+${{ steps.sha.outputs.sha }}")"
-
-      - name: Set version in package.json
-        if: ${{ steps.diff.outputs.count > 0 }}
+      - name: (unstable) Set version in package.json
+        if: ${{ steps.diff.outputs.count > 0 && matrix.version == 'unstable' }}
         run: |
-          echo "$(cat unstable/package.json | jq '.version = "${{ steps.version.outputs.number }}"')" > unstable/package.json
-          echo "$(cat unstable/package-lock.json | jq '.version = "${{ steps.version.outputs.number }}"')" > unstable/package-lock.json
+          SERVER_SHA="$(git ls-remote git://github.com/jellyfin/jellyfin.git | grep refs/heads/master | cut -f 1  | cut -c1-7)"
+          SERVER_VERSION="$(cat openapi.json | jq '.info.version')"
+          SERVER_VERSION=${SERVER_VERSION:1:-1}
+          VERSION=$(echo "$SERVER_VERSION-unstable+$SERVER_SHA")
+          echo "$(cat unstable/package.json | jq ".version = \"${VERSION}\"")" > unstable/package.json
+          echo "$(cat unstable/package-lock.json | jq ".version = \"${VERSION}\"")" > unstable/package-lock.json
```

Comment:
pretty sure `jq` could run into issues with how the envs are used here :thinking: (may work regardless cant recall 4 sure)

based on [this](https://stedolan.github.io/jq/manual/#$ENV,env)